src/Symfony/Component/Serializer/Normalizer/ArrayDenormalizer.php

```diff
@@ -69,8 +73,16 @@ public function supportsDenormalization(mixed $data, string $type, string $forma
             throw new BadMethodCallException(sprintf('The nested denormalizer needs to be set to allow "%s()" to be used.', __METHOD__));
         }
 
-        return str_ends_with($type, '[]')
-            && $this->denormalizer->supportsDenormalization($data, substr($type, 0, -2), $format, $context);
+        if (!str_ends_with($type, '[]')) {
+            return false;
+        }
+
+        if ($this->checkFirstElement) {
+            $data = \is_array($data) ? $data : [];
```

Comment:
Here, if `checkFirstElement` is true and we are giving an `int` for example, we will convert the data to an empty array, which is quite odd to me.
What about returning false in that case?